File: python/tetris/overlay_array.py
def overlay_array(a, b, x=0, y=0):
    for r, row in enumerate(b):
        for c, col in enumerate(row):
            if col != 0:
                if len(a) > y + r and len(a[y + r]) > x + c:
                    a[y + r][x + c] = col

    return a

File: ts/src/overlay-array.ts
export default function overlayArray(
  a: number[][],
  b: number[][],
  x = 0,
  y = 0,
) {
  b.forEach((row, r) => {
    row.forEach((col, c) => {
      if (col) {
        if (a[y + r] === undefined) return;
        if (a[y + r][x + c] === undefined) return;
        a[y + r][x + c] = col;
      }
    });
  });
}
